faiss/utils/distances_fused/simdlib_based.cpp

```diff
@@ -257,17 +257,19 @@ void exhaustive_L2sqr_fused_cmax(
         }
     }
 
-    const size_t nx_p = (nx / NX_POINTS_PER_LOOP) * NX_POINTS_PER_LOOP;
+    const ssize_t nx_p = (ssize_t)((nx / NX_POINTS_PER_LOOP) * NX_POINTS_PER_LOOP);
     // the main loop.
 #pragma omp parallel for schedule(dynamic) num_threads(num_omp_threads)
-    for (size_t i = 0; i < nx_p; i += NX_POINTS_PER_LOOP) {
+    // use signed instead of size_t because of C3016 error on windows:
+    // https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3016
+    for (ssize_t i = 0; i < nx_p; i += (ssize_t)NX_POINTS_PER_LOOP) {
```

Comment:
this works if `ssize_t` is defined and available (it is non standard)
pls refer to facebook/faiss commit https://github.com/facebookresearch/faiss/pull/2889
that fixed the same issue in another file